pallets/stellar-relay/src/lib.rs

```diff
@@ -689,20 +698,20 @@ pub mod pallet {
 		}
 	}
 
-	fn get_memo_hash_from_tx_env(transaction_envelope: &TransactionEnvelope) -> Option<Hash> {
-		let memo_hash = match transaction_envelope {
-			TransactionEnvelope::EnvelopeTypeTxV0(tx_env) => match tx_env.tx.memo {
-				Memo::MemoHash(hash) => Some(hash),
+	fn get_text_hash_from_tx_env(transaction_envelope: &TransactionEnvelope) -> Option<&Vec<u8>> {
```

Comment:
Maybe rename to `get_text_memo_from_tx_env`

Reply:
I changed the name and moved the function to the `primitives` package as this was duplicated code.